libs/api/domains/energy-funds/src/lib/energyFunds.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common'
+import { EnergyFundsClientService } from '@island.is/clients/energy-funds'
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import { VehicleMiniDto, VehicleSearchApi } from '@island.is/clients/vehicles'
+
+@Injectable()
+export class EnergyFundsService {
+  constructor(
+    private readonly energyFundsClientService: EnergyFundsClientService,
+    private readonly vehiclesApi: VehicleSearchApi,
+  ) {}
+
+  private vehiclesApiWithAuth(auth: Auth) {
+    return this.vehiclesApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private getVehicleGrant = async (auth: User, vehicle: VehicleMiniDto) => {
+    return await this.energyFundsClientService.getCatalogValueForVehicle(
+      auth,
+      vehicle,
+    )
+  }
```

Comment:
~ Looks like a unnecessary wrapping as this private function is only used once. So you should just  use `energyFundsClientService` directly.